src/Icicle/Avalanche/Check.hs

```diff
@@ -88,7 +88,13 @@ checkStatement frag ctx stmt
                    $ checkExp frag (ctxExp ctx) x
                Let n x' <$> go stmts
 
-        ForeachInts n from to stmts
+        While t n to stmts
+         -> do to'   <- first ProgramErrorExp
+                      $ checkExp frag (ctxExp ctx) to
+
+               While t n to' <$> go stmts
```

Comment:
oh, I didn't think of that. I did this:

```
        Write n x
         -> do x'   <- first ProgramErrorExp
                     $ checkExp frag (ctxExp ctx) x

               vt   <- maybeToRight (ProgramErrorNoSuchAccumulator n)
                     $ Map.lookup n $ ctxAcc ctx

               let t = annType (annotOfExp x')

               requireSame (ProgramErrorWrongType x) t (FunT [] vt)
               return (Write n x')
```

but I suppose that wouldn't work it `vt` needs to be melted